src/Stream/SourceFilter.h

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+
+#include "FlatbufferMessage.h"
+#include "Metrics/Metric.h"
+#include "Metrics/Registrar.h"
+#include "Stream/MessageWriter.h"
+#include <chrono>
+
+namespace Stream {
+
+using time_point = std::chrono::system_clock::time_point;
+uint64_t toNanoSec(time_point Time);
+
+class SourceFilter {
```

Comment:
Would benefit from a description comment. Particularly to explain the role of the buffered message and connection with the forwarder periodic-update functionality.

Reply:
I have added a short description. You will have to discuss the matter with @mattclarke if it is not to your satisfaction.

Reply:
I think putting the diagram on confluence and adding a link in the comment is a good idea. I also think adding _something_ about the role of `BufferedMessage` is a good idea. Maybe something like this? "SourceFilter buffers a message such that, when used in conjunction with the periodic-update feature of the forwarder, the writer module should always be able to record at least the data from a single message." At least the reader would have some clue what to ask colleagues about to understand what is going on here.

Reply:
If you think it is good to have the diagram on confluence then I think it is better to have it be part of the repository. It is hard to find things there as it is, especially if you do not have access. As for the text, come up with something that both you and @mattclarke agrees on and we an add that.

Reply:
:+1: fine with me, I only suggested confluence because I thought it was dictated that that is where we must put the documentation. I'll talk with Matt about the text.